cmd/examples/server.cpp

```diff
@@ -795,13 +795,21 @@ class MyServer : public quicr::Server
 
         if (groups.empty()) {
             SPDLOG_WARN("No groups found for requested range");
-            return false;
+            return std::nullopt;
         }
 
-        return std::any_of(groups.begin(), groups.end(), [&](const auto& group) {
+        const bool available = std::any_of(groups.begin(), groups.end(), [&](const auto& group) {
             return !group->empty() && group->begin()->headers.object_id <= attrs.start_object &&
                    std::prev(group->end())->headers.object_id >= (attrs.end_object - 1);
         });
+        if (!available) {
+            SPDLOG_WARN("No objects found for requested range");
+            return std::nullopt;
+        }
+        const auto last_cached = groups.back()->end()->headers;
```

Comment:
This will crash, since `end()` is one past the actual end.
```suggestion
        const auto last_cached = std::prev(groups.back()->end())->headers;
```

Reply:
lol I made the exact same mistake on `groups`, fixed it, and then didn't change the one literally right next to it.